### ISSUE TITLE
New failure in 'adaptalint'

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1997]
Haven't had time to explore yet, but there's a new failure in the 'adaptalint' test suite [v3.0.2 vs current HEAD]:

```
── Error (test-basics.R:29:3): extract works in simple case ────────────────────
<tibble_error_column_names_cannot_be_empty/tibble_error/rlang_error/error/condition>
Error in `as_tibble(.)`: Columns 1, 2, 3, 4, 5, and 18 more must be named.
Use `.name_repair` to specify repair.
Caused by error in `repaired_names()`:
! Names can't be empty.
✖ Empty names found at locations 1, 2, 3, 4, 5, etc.
```

It's a simple test of `adaptalint::extract_style()`; that function looks like:

```r
filename %>%
  lint() %>%
  as_tibble() %>%
  mutate(total_lints = n()) %>%
  group_by(.data$linter) %>%
  summarise(count = n(),
            total_lints = mean(.data$total_lints)
  ) %>%
  mutate(adjusted = .data$count/.data$total_lints)
```

The first thing that comes to mind is the new S3 dispatch... we might need to add an `as_tibble.lints` method, perhaps?

**Comment by MichaelChirico** [comment]
Yep... here's the MRE:

```r
tibble::as_tibble(lintr::lint(text="a=1", linters=lintr::assignment_linter()))
```

I think we get to use an `Enhances` dependency, so that's fun :)

**Comment by AshesITR** [comment]
Why doesn't `tibble:::as_tibble.default` get dispatched and returns a good tibble using `lintr:::as.data.frame.lints`?

**Comment by MichaelChirico** [comment]
that's what used to happen, now as_tibble.list intercedes

**Comment by MichaelChirico** [comment]
Maybe there's a `tibble` version consideration here too?

**Comment by AshesITR** [comment]
Ah that makes sense. Can we export an S3 method without importing tibble?
